### PR TITLE
Fix incorrect max_seq_length bug in pytorch training scripts

### LINKED ISSUE #15181: Bug with max_seq_length argument in training scripts
Hello, I am trying to train Nystromformer for Masked Language Modeling with a sequence length of 1024 using the [run_mlm.py](https://github.com/huggingface/transformers/blob/master/examples/pytorch/language-modeling/run_mlm.py) script. As, I understand it, the `max_seq_length` argument of the script controls the maximum number of input tokens passed to the model. And this should be, less than or equal to the model's `max_position_embeddings`. So, I set `--max_seq_length 1024`, however I get: 

`01/17/2022 08:58:11 - WARNING - __main__ - The max_seq_length passed (1024) is larger than the maximum length for themodel (512). Using max_seq_length=512.`

This implies that sequences greater than 512 tokens are not being passed to the model, although the model can handle sequences of upto 1024 tokens (I double-checked that the model has  `max_position_embeddings=1024`).

### To reproduce

1. Clone the transformers repository
2. In the examples/pytorch/language-modeling directory, run `python run_mlm.py --model_type nystromformer --dataset_name wikitext --dataset_config_name wikitext-2-raw-v1 --per_device_train_batch_size 6 --num_train_epochs 1 --do_train --do_eval --tokenizer_name albert-base-v1 --output_dir ./nystrom --max_seq_length 1024 --config_overrides="max_position_embeddings=1024"`

### Potential fix

The `max_seq_length` variable in the script controls the size of the input sequences. [Here](https://github.com/huggingface/transformers/blob/9a2dabae7002258e41419491c73dd43ad61b5de7/examples/pytorch/language-modeling/run_mlm.py#L385), it is set to `min(data_args.max_seq_length, tokenizer.model_max_length)`. However, this will always evaluate to `tokenizer.model_max_length` (if `max_seq_length` is greater than the tokenizer's default `model_max_length`). So, although `--max_seq_length 1024` is passed to the script, the input sequences are truncated to 512 (the default `model_max_length` for tokenizers such as BERT, RoBERTa and ALBERT). 

We could fix this by replacing `tokenizer.model_max_length` with `config.max_position_embeddings` in this [block](https://github.com/huggingface/transformers/blob/0167edc8549b8c8b01fcc669d0da7ea41903b244/examples/pytorch/language-modeling/run_mlm.py#L379-L385). This way, the user is warned if the model cannot support the provided `max_seq_length`, and the sequences are truncated to `min(data_args.max_seq_length, config.max_position_embeddings)`.

I've implemented this change in the pull request below. 

### Who can help
@NielsRogge @LysandreJik @sgugger

### PR DESCRIPTION
**Pull request by novice03**
# What does this PR do?

Fixes #15181 

Input sequences are not correctly truncated to the correct length in the `run_mlm.py` script. Even if a model can support large sequence lengths (such as 1024 or 2048), the sequences are truncated to the tokenizer's default `model_max_length`. This bug is described [here](https://github.com/huggingface/transformers/issues/15181).

This pull request fixes this bug by comparing the `max_seq_length` passed by the user to the maximum sequence length supported by the model (`config.max_position_embeddings`), and sets `max_seq_length = min(data_args.max_seq_length, config.max_position_embeddings)`

If this is the correct fix, I can make the same change in other pytorch training scripts such as [run_plm.py](https://github.com/huggingface/transformers/blob/9a2dabae7002258e41419491c73dd43ad61b5de7/examples/pytorch/language-modeling/run_plm.py#L363), [run_qa.py](https://github.com/huggingface/transformers/blob/9a2dabae7002258e41419491c73dd43ad61b5de7/examples/pytorch/question-answering/run_qa.py#L333), etc.

## Who can review?

@NielsRogge @LysandreJik @sgugger

### CODE REVIEW

**Comment by sgugger** [comment]
As said on the linked issue, this should not be changed.
